trieste/acquisition/rule.py

```diff
@@ -311,124 +302,95 @@ def acquire(
+    """
 
-        If a ``state`` is specified, and the new optimum improves over the previous optimum
-        by some threshold (that scales linearly with ``kappa``), the previous acquisition is
-        considered successful.
+    def __deepcopy__(self, memo: dict[int, object]) -> ContinuousTrustRegionState:
+        box_copy = copy.deepcopy(self.acquisition_space, memo)
+        return ContinuousTrustRegionState(box_copy, self.eps, self.y_min, self.is_global)
 
-        If the previous acquisition was successful, ``search_space`` is used as the new
-        search space. If the previous step was unsuccessful, the search space is changed to the
-        trust region if it was global, and vice versa.
 
-        If the previous acquisition was over the trust region, the size of the trust region is
-        modified. If the previous acquisition was successful, the size is increased by a factor
-        ``1 / beta``. Conversely, if it was unsuccessful, the size is reduced by the factor
-        ``beta``.
+class _ContinuousTrustRegion(EmpiricStateful[ContinuousTrustRegionState, Box]):
+    def __init__(self, global_search_space: Box, beta: float = 0.7, kappa: float = 1e-4):
+        self._global_search_space = global_search_space
+        self._beta = beta
+        self._kappa = kappa
 
-        **Note:** The acquisition search space will never extend beyond the boundary of the
-        ``search_space``. For a local search, the actual search space will be the
-        intersection of the trust region and ``search_space``.
+    @property
+    def default_state(self) -> ContinuousTrustRegionState:
+        space = self._global_search_space
+        eps = 0.5 * (space.upper - space.lower) / (5.0 ** (1.0 / space.lower.shape[-1]))
+        # todo is y_min correct?
+        return ContinuousTrustRegionState(space, eps, math.inf, True)
 
-        :param search_space: The global  :class:`~trieste.space.SearchSpace` for the optimization
-            problem.
-        :param datasets: The known observer query points and observations. Uses the data for key
-            `OBJECTIVE` to calculate the new trust region.
-        :param models: The models of the specified ``datasets``.
-        :param state: The acquisition state from the previous step, if there was a previous step,
-            else `None`.
-        :return: A 2-tuple of the query point and the acquisition state for this step.
-        :raise KeyError: If ``datasets`` does not contain the key `OBJECTIVE`.
-        """
+    def acquire(
+        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
+    ) -> State[ContinuousTrustRegionState, Box]:
         dataset = datasets[OBJECTIVE]
 
-        global_lower = search_space.lower
-        global_upper = search_space.upper
+        global_lower = self._global_search_space.lower
+        global_upper = self._global_search_space.upper
 
         y_min = tf.reduce_min(dataset.observations, axis=0)
```

Comment:
@vpicheny I chose the starting `y_min` to be infinite. However, this isn't right if `dataset.observations` is empty. What should happen if `dataset.observations` is empty? Should trust region error?

Reply:
I think this is fine. If you start with y_min=inf, your first step will always be a success, so you'll start by doing at least two global steps, which is reasonable in absence of previous data.

Reply:
i made it error if `y_min` is not finite

Reply:
ah, i didn't see your message. If `state.y_min` is infinite, and `y_min` is also infinite, you'll go straight to a local search space because `y_min < state.y_min - self._kappa * tr_volume` and `not state.is_global` are both false